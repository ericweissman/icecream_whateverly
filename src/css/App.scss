.App {
  text-align: center;
  position: relative;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.card-container {
  display: flex;
  flex-flow: wrap;

  
}



.header {
  font-size: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: aquamarine;
  margin-bottom: 25px;
  padding: 50px;
}

