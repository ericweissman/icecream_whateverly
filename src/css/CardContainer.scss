.card-container {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr; 
    background-color: black;
    margin: auto;
  }
