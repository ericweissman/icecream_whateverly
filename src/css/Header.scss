@import url('https://fonts.googleapis.com/css?family=Lobster');

.header {
    display: flex;
    flex-direction: column;
    font-size: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: aquamarine;
    padding: 50px;
}

.title {
    font-family: 'Lobster', cursive;
    font-size: 75px;
}
