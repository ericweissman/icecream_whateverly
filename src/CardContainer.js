import React, { Component } from 'react';
import App from './App.js';
import './css/Main.scss';
import { uid } from 'react-uid'
import Card from './Card.js'

class CardContainer extends Component {
  constructor(props) {
    super(props)
  };

  

  render() {
    const keys = Object.keys(this.props.icecream);
    return (
      <div className="card-container">
        {
          keys.map((key) => {
            const flavor = this.props.icecream[key].name.toLowerCase()
            if (flavor.includes(this.props.search.toLowerCase())) {
              return (
                <Card 
                  id={key}
                  img={key}
                  flavor={this.props.icecream[key].name}
                  info={this.props.icecream[key].description}
                  parlors={this.props.parlors}
                />
            )}
          })
        }
      </div>
    )
  }
}
export default CardContainer;