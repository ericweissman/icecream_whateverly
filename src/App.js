import React, { Component } from 'react';
import CardContainer from './CardContainer.js';
import Header from './Header';
import Banner from './Banner';
import './css/Main.scss';
import SearchByParlor from './SearchByParlor';

class App extends Component {
  constructor() {
    super();
    
    this.state = {
      parlors: [],
      icecream: {},
      value: '',
      searchByParlor: ''
    }
    
  }
  componentDidMount() {
    fetch("https://whateverly-datasets.herokuapp.com/api/v1/parlors")
      .then(results => results.json())
      .then((result) => {
        this.setState({
          parlors: result.parlors
        })
      })
      .catch(err => {
        this.setState({error: err})
      })

    fetch("https://whateverly-datasets.herokuapp.com/api/v1/flavors")
      .then(results => results.json())
      .then((result) => {
        this.setState({
          icecream: result.flavors
        })
      })
      .catch(err => {
        this.setState({error: err})
      })
  }

  searchFor = (event) => {
    this.setState({value: event.target.value})
  }

  searchParlor = (event) => {
    this.setState({searchByParlor: event.target.value})
  }

  render() {
    return (
      <div className="App">
        <div>
          <Header 
            searchFor={this.searchFor}
            searchParlor={this.searchParlor} 
            parlors={this.state.parlors}
          />
          <Banner />
          <CardContainer parlors={this.state.parlors} 
                        icecream={this.state.icecream}
                        search={this.state.value}
                        searchByParlor={this.state.searchByParlor}/>
          {/* <ErrorMessage /> */}
        </div>
      </div>
    );
  }
} 

export default App;