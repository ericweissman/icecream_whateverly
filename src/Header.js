import React, { Component } from 'react';
import App from './App.js';
import './css/Main.scss';
import { uid } from 'react-uid';
import Card from './Card.js';
import Search from './Search.js';


class Header extends Component {
  constructor() {
    super();
    }
    render() {
      return (
        <header className="header">
          <div className="header-svg-cont">
            <img className="header-svg" src={require('./css/images/ice-cream.svg')}/>
          </div>
          <div className="header-search-cont">
            <Search searchFor={this.props.searchFor}/>
          </div>
        </header>
      );
    }
  }



export default Header;