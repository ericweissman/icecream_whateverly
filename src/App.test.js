import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import Header from './Header';
import CardContainer from './CardContainer';

const searchForMock = jest.fn();
const parlors = [{
  "parlorName": "Little Man Ice Cream",
  "description": "Quirky shop for exotic ice cream",
  "address": "2620 16th St, Denver, CO 80211",
  "daysOpen": [
    {
      "day": "Sunday",
      "open": "11 AM",
      "close": "10 PM"
    },
    {
      "day": "Monday",
      "open": "11 AM",
      "close": "10 PM"
    },
    {
      "day": "Tuesday",
      "open": "11 AM",
      "close": "10 PM"
    },
    {
      "day": "Wednesday",
      "open": "11 AM",
      "close": "10 PM"
    },
    {
      "day": "Thursday",
      "open": "11 AM",
      "close": "10 PM"
    },
    {
      "day": "Friday",
      "open": "11 AM",
      "close": "11 PM"
    },
    {
      "day": "Saturday",
      "open": "11 AM",
      "close": "11 PM"
    }
  ],
  "phoneNumber": "303-455-3811",
  "website": "http://www.littlemanicecream.com/latest-scoop/menu/",
  "flavors": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27],
  "numberOfFlavors": 27,
  "priceRange": "$",
  "review": 4.7
}]
const icecream = {
  "1": {
    "name": "Vanilla",
    "description": "An old fashioned vanilla ice cream recipe that everyone loves"
  }
}
const value = ''

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <div>
      <Header searchFor={searchForMock} />
      <CardContainer parlors={parlors}
        icecream={icecream}
        search={value} />
        </div>
    );
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual({ value: ''})
  })
})