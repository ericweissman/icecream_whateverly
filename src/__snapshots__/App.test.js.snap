// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<div>
  <Header
    searchFor={[MockFunction]}
  />
  <CardContainer
    icecream={
      Object {
        "1": Object {
          "description": "An old fashioned vanilla ice cream recipe that everyone loves",
          "name": "Vanilla",
        },
      }
    }
    parlors={
      Array [
        Object {
          "address": "2620 16th St, Denver, CO 80211",
          "daysOpen": Array [
            Object {
              "close": "10 PM",
              "day": "Sunday",
              "open": "11 AM",
            },
            Object {
              "close": "10 PM",
              "day": "Monday",
              "open": "11 AM",
            },
            Object {
              "close": "10 PM",
              "day": "Tuesday",
              "open": "11 AM",
            },
            Object {
              "close": "10 PM",
              "day": "Wednesday",
              "open": "11 AM",
            },
            Object {
              "close": "10 PM",
              "day": "Thursday",
              "open": "11 AM",
            },
            Object {
              "close": "11 PM",
              "day": "Friday",
              "open": "11 AM",
            },
            Object {
              "close": "11 PM",
              "day": "Saturday",
              "open": "11 AM",
            },
          ],
          "description": "Quirky shop for exotic ice cream",
          "flavors": Array [
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            9,
            10,
            11,
            12,
            13,
            14,
            15,
            16,
            17,
            18,
            19,
            20,
            21,
            22,
            23,
            24,
            25,
            26,
            27,
          ],
          "numberOfFlavors": 27,
          "parlorName": "Little Man Ice Cream",
          "phoneNumber": "303-455-3811",
          "priceRange": "$",
          "review": 4.7,
          "website": "http://www.littlemanicecream.com/latest-scoop/menu/",
        },
      ]
    }
    search=""
  />
</div>
`;
